examples/getstarted/src/plugins/mycustomfields/admin/src/index.js

```diff
@@ -41,6 +42,83 @@ export default {
               /* webpackChunkName: "input-component" */ './components/ColorPicker/ColorPickerInput'
             ),
         },
+        options: {
+          base: [
+            {
+              intlLabel: {
+                id: 'color-picker.color.format.label',
+                defaultMessage: 'Color format',
+              },
+              name: 'options.color-picker.format',
+              type: 'select',
+              value: 'hex',
+              options: [
+                {
+                  key: 'hex',
+                  value: 'hex',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.hex',
+                      defaultMessage: 'Hexadecimal',
+                    },
+                  },
+                },
+                {
+                  key: 'rgba',
+                  value: 'rgba',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.rgba',
+                      defaultMessage: 'RGBA',
+                    },
+                  },
+                },
+              ],
+            },
+          ],
+          advanced: [
+            {
+              sectionTitle: {
+                id: 'global.settings',
+                defaultMessage: 'Settings',
+              },
+              items: [
+                {
+                  name: 'required',
+                  type: 'checkbox',
+                  intlLabel: {
+                    id: 'form.attribute.item.requiredField',
+                    defaultMessage: 'Required field',
+                  },
+                  description: {
+                    id: 'form.attribute.item.requiredField.description',
+                    defaultMessage: "You won't be able to create an entry if this field is empty",
+                  },
+                },
+                {
+                  name: 'private',
+                  type: 'checkbox',
+                  intlLabel: {
+                    id: 'form.attribute.item.privateField',
+                    defaultMessage: 'Private field',
+                  },
+                  description: {
+                    id: 'form.attribute.item.privateField.description',
+                    defaultMessage: 'This field will not show up in the API response',
+                  },
+                },
+              ],
+            },
+          ],
+          validator: args => ({
+            'color-picker': yup.object().shape({
+              format: yup.string().required({
+                id: 'options.color-picker.format.error',
+                defaultMessage: 'The color format is required',
+              }),
+            }),
+          }),
```

Comment:
I've mentioned it on Slack already and don't want to bikeshed here, so please feel free to ignore if you think it is not a good idea:

1. I'd rename this method to `validate()`, to make it more imperative
2. I'd change the signature to return a `boolean` instead, because it might feel more intuitive to our users. They'd still have the chance to validate the `yup` schema.

Reply:
I think your point is valid, but after looking a bit more at this and discussing with Rémi I don’t think we will make this change for a few reasons:

1. This design was detailed in the RFC which was left in review for over a month
2. We couldn’t return just true or false, we would need the name of the input and an error message as well. Throwing a general error message is not a great UX.
3. Forcing a user to use Yup is not necessarily a bad thing. It is consistent with how validation is used in Strapi and it is not a difficult API to use.

For the naming you also make a good argument but we will also stick with `validator` since we are using an existing API, the content-type builder’s forms API, that has already established this convention.